tensorflow/python/kernel_tests/conv2d_transpose_test.py

```diff
@@ -157,6 +158,129 @@ def testGradient(self):
     err_tolerance = 0.0005
     self.assertLess(err, err_tolerance)
 
+  def get_available_gpus(self):
+    local_device_protos = device_lib.list_local_devices()
+    return [x.name for x in local_device_protos if x.device_type == 'GPU']
+
+  def testConv2DTransposeSingleStrideNCHW(self):
+    # `NCHW` data fomat is only supported for `GPU` device.
+    if len(self.get_available_gpus()) < 1:
```

Comment:
Yeah, your observation is expected. test_session by default put the op on CPU; you can use self.test_session(use_gpu=True) to change this default setting.